### PR TITLE
ci: tag all infra images with proper prefix

### PR DESCRIPTION
**Pull request by sdowell**
The infrastructure-public-image prefix advertises that these images are intended for use by end to end testing infrastrucutre.

### CODE REVIEW

**Review by janetkuo — APPROVED** [review]
/lgtm

File: Makefile.e2e.ci [hunk 1]
```diff
@@ -8,11 +8,18 @@
 test-e2e-gke-ci: pull-postsubmit-retry test-e2e-gke-nobuild
 
 POSTSUBMIT_GCS_PREFIX ?= gs://kpt-config-sync-ci-postsubmit
-POSTSUBMIT_REGISTRY ?= us-docker.pkg.dev/kpt-config-sync-ci-artifacts/postsubmit
+POSTSUBMIT_REGISTRY ?= $(LOCATION)-docker.pkg.dev/$(TEST_INFRA_PROJECT)/postsubmit
 
 .PHONY: postsubmit
-postsubmit: build-cli
```
Comment (nan-yu) [inline Makefile.e2e.ci:14]: I think `postsubmit` still depends on `build-cli` as `publish-gcs` publishes both manifests and nomos CLI.

Comment (sdowell) [reply Makefile.e2e.ci:14]: Yeah this was an accident that got left in from local testing, see https://github.com/GoogleContainerTools/kpt-config-sync/pull/1163

**Pull request merged**